x/airdrop/types/airdrop.go

```diff
@@ -28,16 +28,17 @@ func (a *Airdrop) GetCurrentDateIndex(ctx sdk.Context, windowLengthSeconds int64
 	elapsedDays := elapsedTimeSeconds / windowLengthSeconds
 
 	// Cap the airdrop index at the last day
-	if elapsedDays > a.GetAirdropLength() {
-		elapsedDays = a.GetAirdropLength() - 1
+	if elapsedDays > a.GetAirdropLength(windowLengthSeconds) {
+		elapsedDays = a.GetAirdropLength(windowLengthSeconds) - 1
 	}
 
 	return int(elapsedDays), nil
 }
 
 // Returns the length of the airdrop in days
-func (a *Airdrop) GetAirdropLength() int64 {
+// windowLengthSeconds is the time between each element in the allocations array
+func (a *Airdrop) GetAirdropLength(windowLengthSeconds int64) int64 {
```

Comment:
Nice, this makes a lot of sense, and I think this is an overall cleaner design! 

This is outside the scope of this PR I think, but would it make more sense to relabel "days" to "periods" with this new structure?

Reply:
agreed! 

Also I don't really love the naming of `windowLengthSeconds` or `allocation_window_seconds` - very open to better names if you can think of any!

Reply:
just renamed `GetAirdropLength` to `GetAirdropPeriods`

Reply:
Hm how about `periodLengthSeconds` and `period_length_seconds`? Open to other options too!

Reply:
I like it!

Reply:
done https://github.com/Stride-Labs/stride/pull/1235/commits/84abb9c5517b6f74b887c5d3fcbbf2ea89245330

Reply:
Incredible, thank you for doing that!